samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -606,118 +608,66 @@ static struct bt_bap_stream_ops stream_ops = {
+ * Gets BIS channel allocation (if exists) and updates the bis_channel_allocation_data struct
+ * Always returns `true` to continue to next BIS
  */
-static bool find_valid_bis_cb(const struct bt_bap_base_subgroup_bis *bis,
-					       void *user_data)
+static bool bis_get_channel_allocation_cb(const struct bt_bap_base_subgroup_bis *bis,
+					  void *user_data)
 {
-	struct find_valid_bis_data *data = user_data;
-	struct bt_audio_codec_cfg codec_cfg = {0};
+	struct bis_channel_allocation_data *data = user_data;
+	struct bt_audio_codec_cfg codec_cfg;
 	enum bt_audio_location chan_allocation;
-	int err;
+	int err, idx;
+
+	idx = data->cnt++;
+	data->bis[idx].index = bis->index;
 
 	err = bt_bap_base_subgroup_bis_codec_to_codec_cfg(bis, &codec_cfg);
 	if (err != 0) {
 		printk("Could not get codec configuration for BIS: %d\n", err);
-		return true;
+		data->bis[idx].chan_allocation_available = false;
+
+		return true; /* continue to next BIS */
 	}
 
+	/* bt_bap_base_subgroup_bis_codec_to_codec_cfg only updates the codec specific capabilities
+	 * data. The bap_base_codec_id data in the codec_cfg struct needed by
+	 * bt_audio_codec_cfg_get_chan_allocation for the fallback check.
+	 */
+	codec_cfg.id = data->bap_base_codec_id.id;
+	codec_cfg.cid = data->bap_base_codec_id.cid;
+	codec_cfg.vid = data->bap_base_codec_id.vid;
+
+	/* Fallback: If no channel allocation data present then channel allocation is set to MONO if
+	 * codec id is BT_HCI_CODING_FORMAT_LC3
+	 */
 	err = bt_audio_codec_cfg_get_chan_allocation(&codec_cfg, &chan_allocation, true);
 	if (err != 0) {
 		printk("Could not find channel allocation for BIS: %d\n", err);
+		data->bis[idx].chan_allocation_available = false;
 
-		if (err == -ENODATA && strlen(CONFIG_TARGET_BROADCAST_NAME) > 0U) {
-			/* Accept no channel allocation data available
-			 * if TARGET_BROADCAST_NAME defined. Use current index.
-			 */
-			data->bis[0].index = bis->index;
-			data->bis[0].chan_allocation = chan_allocation;
-			data->cnt = 1;
-
-			return false;
-		}
-	} else {
-		if ((chan_allocation & CONFIG_TARGET_BROADCAST_CHANNEL) ==
-		    CONFIG_TARGET_BROADCAST_CHANNEL) {
-			/* Found single BIS with all channels we want - keep as only and stop
-			 * parsing
-			 */
-			data->bis[0].index = bis->index;
-			data->bis[0].chan_allocation = chan_allocation;
-			data->cnt = 1;
-
-			return false;
-		} else if ((chan_allocation & CONFIG_TARGET_BROADCAST_CHANNEL) != 0) {
-			/* BIS contains part of what we are looking for - Store and see if there are
-			 * other BIS that may fill the gaps
-			 */
-			data->bis[data->cnt].index = bis->index;
-			data->bis[data->cnt].chan_allocation = chan_allocation;
-			data->cnt++;
-		}
+		return true; /* continue to next BIS */
 	}
 
-	return true;
-}
+	/* Store channel allocation data for this bis */
+	data->bis[idx].chan_allocation = chan_allocation;
+	data->bis[idx].chan_allocation_available = true;
```

Comment:
What does `chan_allocation_available` represent here? 

Since you are using fallback above, then this will be set to `true` regardless of whether there's a location bit set for the BIS.

Reply:
Correct. But if fallback is not used this will ensure that subgroup allocation is used for this BIS. Also the case if bt_bap_base_subgroup_bis_codec_to_codec_cfg somehow fails